cf-agent/verify_processes.c

```diff
@@ -94,9 +96,24 @@ static bool ProcessSanityChecks(const Attributes *a, const Promise *pp)
     {
         Log(LOG_LEVEL_ERR, "Process select constraint body promised no result (check body definition)");
         PromiseRef(LOG_LEVEL_ERR, pp);
-        return false;
+        ret = false;
     }
 
+    if ((a->haveprocess_count) && ((!a->process_count.min_range) && (!a->process_count.max_range)))
+    {
+        Log(LOG_LEVEL_ERR, "process_count used without match_range, this promise will not define a class based on a count of matching promises.");
+        PromiseRef(LOG_LEVEL_ERR, pp);
+        ret = false;
+    }
+
+    if ((a->haveprocess_count) && ((!a->process_count.in_range_define) || (!a->process_count.out_of_range_define)))
+    {
+        Log(LOG_LEVEL_ERR, "process_count body is insufficiently defined. process_count must specify at least one of in_range_define or out_of_range_define. ");
```

Comment:
It seems like MPF does this incorrectly then? 

```
Q: ".../cf-agent" --de":    error: Promise belongs to bundle 'cfe_internal_limit_robot_agents_reap' in file '/var/cfengine/inputs/cfe_internal/core/limit_robot_agents.cf' near line 39
Q: ".../cf-agent" --de":    error: Comment is 'Check cf-monitord process if exceed the **number'**
```

This change causes many error messages in the valgrind / bootstrap test. See Travis logs for more details. Please fix these in MPF before merging, if you believe the error message is correct.

Reply:
Oh, this is bad logic on my part. I want to error when neither in_range_define nor out_of_range_define is specified.  Basically, in order to have a useful process_count body, you must have match_range and you must have one of either out_of_range_define or in_range_define.


Thanks